lib/utils/retry.go

```diff
@@ -49,6 +54,25 @@ func NewJitter() Jitter {
 	}
 }
 
+// NewSmallJitter builds a new small jitter (currently jitters on
+// the range [6n/7,n), but this is subject to change).  Prefer this over
+// NewJitter() when jittering periodic operations (e.g. cert rotation
+// checks) since large jitters result in significantly increased load.
+func NewSmallJitter() Jitter {
+	var mu sync.Mutex
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	return func(d time.Duration) time.Duration {
+		// values less than 1 cause rng to panic, and some logic
+		// relies on treating zero duration as non-blocking case.
+		if d < 1 {
+			return 0
+		}
+		mu.Lock()
+		defer mu.Unlock()
+		return (6 * d / 7) + time.Duration(rng.Int63n(int64(d))/7)
```

Comment:
What do these constants represent?

Reply:
In terms of what the math is doing, its randomizing up to one seventh of the supplied duration (i.e. creating a jitter on the range of `[6n/7,n)`).

In terms of *why* I chose that specific constant, its essentially arbitrary.  We needed a smaller jitter than `1/2` (what the default jitter uses) for periodic operations since a jitter of `1/2` could worst-case double the amount of work.  I went with a `1/7` jitter because most of our periodic operations happen on a 10 minute interval, and a `1/7` jitter will worst-case cause 1 additional operation per hour when dealing with 10-minute intervals.  This is itself a pretty arbitrary goal, but it seemed reasonable.  It means that if instances did sync up coincidentally, you'd expect them to be reasonably "desynced" within an hour.  They won't be fully desynced, since repeated jitters tend to scatter on a normal distribution, but desynced enough to have a standard deviation of 1 minute.

Reply:
Can you push choosing these fractions up to the caller?
Something like `func NewJitter(frac float64) Jitter // returns values between [(1-frac) * n : n)`, which can then be caller like `NewJitter(0.5)`

Reply:
Tried a few different iterations of having caller supply fraction manually.  Everything I came up with felt a bit too brittle.  Instead, I've renamed the `DefaultJitter` to `HalfJitter` and `NewSmallJitter` to `NewSeventhJitter` so that the caller now explicitly commits to the desired fraction by selecting the appropriate function.  Ex:

```golang
interval.New(interval.Config{
    Duration:      defaults.HighResPollingPeriod,
    FirstDuration: utils.HalfJitter(defaults.HighResPollingPeriod),
    Jitter:        utils.NewSeventhJitter(),
})
```